convert.c

```diff
@@ -256,6 +257,121 @@ static int will_convert_lf_to_crlf(size_t len, struct text_stat *stats,
 
 }
 
+static struct encoding {
+	const char *name;
+	iconv_t to_git;       /* conversion to UTF-8 encoding */
+	iconv_t to_worktree;  /* conversion to user-defined encoding */
+	struct encoding *next;
+} *encoding, **encoding_tail;
+static const char *default_encoding = "utf-8";
+static iconv_t no_conversion = (iconv_t)-1;
+
+static int encode_to_git(const char *path, const char *src, size_t src_len,
+			 struct strbuf *buf, struct encoding *enc)
+{
+#ifndef NO_ICONV
+	char *dst, *re_src;
+	int dst_len, re_src_len;
+
+	/*
+	 * No encoding is specified or there is nothing to encode.
+	 * Tell the caller that the content was not modified.
+	 */
+	if (!enc || (src && !src_len))
+		return 0;
+
+	/*
+	 * Looks like we got called from "would_convert_to_git()".
+	 * This means Git wants to know if it would encode (= modify!)
+	 * the content. Let's answer with "yes" since an encoding was
+	 * specified.
+	 */
+	if (!buf && !src)
+		return 1;
+
+	if (enc->to_git == no_conversion) {
```

Comment:
I’m thinking that `no_conversion` might be a bit misleading. For instance, `-1` is returned upon errors by `iconv_open`, which doesn’t imply that “no conversion” is intended.

Another proposal would be `invalid_conversion`. Maybe something like `iconv_t_invalid` might be even more concise (as this really just is an `iconv_t` value and nothing else), however, this might clash with future naming decisions in upstream `iconv`.

Reply:
`invalid_conversion` 👍